devito/passes/iet/mpi.py

```diff
@@ -368,3 +338,62 @@ def mpiize(graph, **kwargs):
         make_halo_exchanges(graph, mpimode=mpimode, **kwargs)
 
     make_reductions(graph, mpimode=mpimode, **kwargs)
+
+
+# *** Utilities
+
+def _filter_iter_mapper(iet):
+    """
+    Given an IET, return a mapper from Iterations to the HaloSpots.
+    Additionally, filter out Iterations that are not of interest.
+    """
+    iter_mapper = {}
+    for k, v in MapNodes(Iteration, HaloSpot, 'immediate').visit(iet).items():
+        filtered_hs = [hs for hs in v if not hs.halo_scheme.is_void]
+        if k is not None and len(filtered_hs) > 1:
+            iter_mapper[k] = filtered_hs
+
+    return iter_mapper
+
+
+def _make_cond_mapper(iet):
+
+    cond_mapper = {}
+    for hs, v in MapHaloSpots().visit(iet).items():
+        conditionals = set()
+        for i in v:
+            if i.is_Conditional and not isinstance(i.condition, GuardFactorEq):
```

Comment:
this smells like a hack. What are you trying to accomplish here?

Is this a recent addition or is it something I've missed all along until now?

also, it should have been a set comprehension --

```
conditionals = {i for i in v if i.is_Conditional ...}
```

But again, this line seems a bit dodgy to me